driller/driller.py

```diff
@@ -140,101 +111,64 @@ def _drill_input(self):
         p = angr.misc.tracer.make_tracer_project(binary=self.binary, hooks=self._hooks)
         s = p.factory.tracer_state(input_content=self.input, magic_content=r.magic)
 
-        self._set_concretizations(s)
-
         simgr = p.factory.simgr(s, save_unsat=True, hierarchy=False, save_unconstrained=r.crash_mode)
 
         t = angr.exploration_techniques.Tracer(trace=r.trace)
         c = angr.exploration_techniques.CrashMonitor(trace=r.trace, crash_mode=r.crash_mode, crash_addr=r.crash_addr)
+        self._core = angr.exploration_techniques.DrillerCore(trace=r.trace)
 
         simgr.use_technique(c)
         simgr.use_technique(t)
         simgr.use_technique(angr.exploration_techniques.Oppologist())
+        simgr.use_technique(self._core)
 
-        # update encounters with known state transitions
-        self._encounters.update(izip(r.trace, islice(r.trace, 1, None)))
+        self._set_concretizations(simgr.one_active)
 
-        l.debug("drilling into %r", self.input)
-        l.debug("input is %r", self.input)
+        l.debug("Drilling into %r.", self.input)
+        l.debug("Input is %r.", self.input)
 
-        # used for finding the right index in the fuzz_bitmap
-        prev_loc = 0
+        simgr.run()
```

Comment:
I think we should still do the loop of stepping and so forth. That way, we can a) yield diverged inputs as we go and b) maintain that redis check just in case anyone actually uses that stuff.